src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -370,6 +370,21 @@ def submit_pipeline(
                 args_for_step_executor.setdefault(
                     "max_run", step_settings.max_runtime_in_seconds
                 )
+
+                # Add training image config if training_repository_access_mode is configured
+                if step_settings.training_repository_access_mode:
+                    training_image_config: Dict[str, Any] = {
+                        "TrainingRepositoryAccessMode": step_settings.training_repository_access_mode.value
+                    }
+                    if step_settings.training_repository_credentials_provider_arn:
+                        training_image_config[
+                            "TrainingRepositoryAuthConfig"
+                        ] = {
+                            "TrainingRepositoryCredentialsProviderArn": step_settings.training_repository_credentials_provider_arn
+                        }
+                    args_for_step_executor.setdefault(
```

Comment:
All of this was already possible by including it in this general-purpose attribute `estimator_args`, correct? Do we really need to maintain this code for such a highly specialized case, if it was already possible? Your code doesn't simply ignores these options now if different values are being passed in `estimator_args`